ui.tests/test-module/specs/checkboxgroup/checkboxgroup.runtime.spec.js

```diff
@@ -0,0 +1,96 @@
+    const selectors = {
+        checkboxgroup : `[data-cmp-is="${IS}"]`
+    }
+
+    let formContainer = null
+
+    beforeEach(() => {
+        cy.previewForm(pagePath).then(p => {
+            formContainer = p;
+        })
+    });
+
+    const checkHTML = (id, state) => {
+        const visible = state.visible;
+        const passVisibleCheck = `${visible === true ? "" : "not."}be.visible`;
+        const passDisabledAttributeCheck = `${state.enabled === false ? "" : "not."}have.attr`;
+        const value = state.value
+        cy.get(`#${id}`)
+            .should(passVisibleCheck)
+            .invoke('attr', 'data-cmp-visible')
+            .should('eq', visible.toString());
+        cy.get(`#${id}`)
+            .invoke('attr', 'data-cmp-enabled')
+            .should('eq', state.enabled.toString());
+        return cy.get(`#${id}`).within((root) => {
+            cy.get('*').should(passVisibleCheck)
+            cy.get('input')
+                .should('have.length', 4)
+            cy.get('input')
+                .should(passDisabledAttributeCheck, 'disabled');
+            cy.get('input').eq(1).should('be.checked')
+        })
+    }
+
+    it(" should get model and view initialized properly ", () => {
+        expect(formContainer, "formcontainer is initialized").to.not.be.null;
+        expect(formContainer._model.items.length, "model and view elements match").to.equal(Object.keys(formContainer._fields).length);
+        Object.entries(formContainer._fields).forEach(([id, field]) => {
+            expect(field.getId()).to.equal(id)
+            expect(formContainer._model.getElement(id), `model and view are in sync`).to.equal(field.getModel())
+        });
+    })
+
+    it(" model's changes are reflected in the html ", () => {
+        const [id, fieldView] = Object.entries(formContainer._fields)[0]
+        const model = formContainer._model.getElement(id)
+        const val = Array('1','2')
+        model.value = '1'
+        checkHTML(model.id, model.getState()).then(() => {
+            model.visible = false
+            return checkHTML(model.id, model.getState())
+        }).then(() => {
+            model.enable = false
+            return checkHTML(model.id, model.getState())
+        })
+    });
+
+    it(" html changes are reflected in model ", () => {
+        const [id, fieldView] = Object.entries(formContainer._fields)[0]
+        const model = formContainer._model.getElement(id)
+
+        cy.get(`#${id}`).find("input").eq(1).click().then(x => {
+            let mval = model.getState().value
+            cy.wrap(model.getState().value).each(($el, index) => {
+                expect($el).to.include(mval[index])
+            })
+        })
+        cy.get(`#${id}`).find("input").eq(2).click().then(x => {
+            let mval = model.getState().value
+            cy.wrap(model.getState().value).each(($el, index) => {
+                expect($el).to.include(mval[index])
+            })
+        })
+    });
```

Comment:
can you add test cases for error messages as well. It is missing in textinput and I was not able to do it. I will add later in textinput, but please add for checkbox.

Reply:
Added the test